airflow/timetables/base.py

```diff
@@ -83,6 +83,13 @@ def interval(cls, start: DateTime, end: DateTime) -> "DagRunInfo":
 class Timetable(Protocol):
     """Protocol that all Timetable classes are expected to implement."""
 
+    @property
+    def interval_description(self) -> [str, None]:
```

Comment:
```suggestion
    def interval_description(self) -> Optional[str]:
```

Or we can make this `str` and return an empty string when a description is not available, since an empty description does not make sense in the UI anyway. Instead of `if dag.timetable.interval_description is not None` we can just do `if not dag.timetable.interval_description`.

Reply:
Agree, but I think Optional[] is more descriptive, as it means that there is no description available rather.